analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/TestAnalysisComparer.cs

```diff
@@ -0,0 +1,132 @@
+                {
+                    summary.Append($"Line {expectedIssue.LineNumber}: expected {IssueType(expectedIssue.IsPrimary)} issue {expectedIssue.IssueId ?? "(no id)"} was not raised!\n");
+                }
+            }
+
+            return summary;
+        }
+
+        private static StringBuilder Compare(IIssueLocation actual, IIssueLocation expected)
+        {
+            var summary = new StringBuilder();
+            var prefix = $"Line {actual.LineNumber}";
+            var issueType = IssueType(actual.IsPrimary);
+            var issueId = expected?.IssueId ?? actual.IssueId ?? "(no id)";
+
+            if (expected == null)
+            {
+                summary.Append($"{prefix}: unexpected {issueType} issue {issueId} with message '{actual.Message}'.\n");
+            }
+            else if (expected.Message != null && actual.Message != expected.Message)
+            {
+                summary.Append($"{prefix}: {issueType} issue {issueId} message '{expected.Message}' does not match the actual message '{actual.Message}'.\n");
+            }
+            else if (expected.Start.HasValue && actual.Start != expected.Start)
+            {
+                summary.Append($"{prefix}: {issueType} issue {issueId} should start on column {expected.Start} but got column {actual.Start}.\n");
+            }
+            else if (expected.Length.HasValue && actual.Length != expected.Length)
+            {
+                summary.Append($"{prefix}: {issueType} issue {issueId} should have a length of {expected.Length} but got a length of {actual.Length}.\n");
+            }
+
+            return summary;
+        }
+
+        private static IIssueLocation ToIssueLocation(this Diagnostic diagnostic) =>
+            new IssueLocationCollector.IssueLocation
+            {
+                IsPrimary = true,
+                LineNumber = diagnostic.Location.GetLineNumberToReport(),
+                Message = diagnostic.GetMessage(),
+                IssueId = diagnostic.Id,
+                Start = diagnostic.Location.GetLineSpan().StartLinePosition.Character,
+                Length = diagnostic.Location.SourceSpan.Length
+            };
+
+        private static IIssueLocation ToIssueLocation(this SecondaryLocation secondaryLocation) =>
+            new IssueLocationCollector.IssueLocation
+            {
+                IsPrimary = false,
+                LineNumber = secondaryLocation.Location.GetLineNumberToReport(),
+                Message = secondaryLocation.Message,
+                IssueId = null,
+                Start = secondaryLocation.Location.GetLineSpan().StartLinePosition.Character,
+                Length = secondaryLocation.Location.SourceSpan.Length
+            };
+
+        private static Dictionary<string, IList<IIssueLocation>> ToIssueLocations(this IEnumerable<Diagnostic> diagnostics)
+        {
+            var actualIssues = new Dictionary<string, IList<IIssueLocation>>();
+
```

Comment:
This method can be without empty lines. There is no major unexpected stuff going on

Reply:
For me it's much more easy to read the code if I have some spaces around the `foreach` and the `if` branches.